src/Expander.hs

```diff
@@ -558,18 +553,14 @@ initializeKernel outputChannel = do
               ValueIOAction $ do
                 vx <- mx
                 vioy <- case f of
-                  HO fun -> do
-                    pure (fun vx)
+                  HO fun -> pure (fun vx)
                   FO clos -> do
                     let env = view closureEnv clos
                         var = view closureVar clos
                         ident = view closureIdent clos
                         body = view closureBody clos
-                    out <- runExceptT $ flip runReaderT env $ runEval $
-                      withExtendedEnv ident var vx $
-                        eval body
-                    case out of
-                      Left err -> error (T.unpack (pretty err))
+                    case (evaluateWithExtendedEnv env [(ident, var, vx)] body) of
+                      Left err -> error (T.unpack (pretty $ projectError err))
```

Comment:
woah! why do we fail with `error` here? I think `runEval` used to only fail if there was a bug in our implementation of the compiler, e.g. if our type checker allowed the construction of an ill-typed Core term. but nowadays evaluating a Core expression can fail with `EvalErrorCase` or `EvalErrorUser` if there is a bug in the user's `.kl` code too, so we should probably use `throwError` here. this is unrelated to the current PR, so I'll create a separate ticket to track this: #237